packages/conf-rust-2021/conf-rust-2021.1/opam

```diff
@@ -0,0 +1,34 @@
+opam-version: "2.0"
+maintainer: "contact@trili.tech"
+authors: [
+  "TriliTech"
+  "Simon Cruanes"
+]
+homepage: "https://github.com/ocaml/opam-repository"
+bug-reports: "https://github.com/ocaml/opam-repository/issues"
+license: "MIT"
+extra-files: ["test.rs" "sha256=536e506bb90914c243a12b397b9a998f85ae2cbd9ba02dfd03a9e155ca5ca0f4"]
+build: [
+  ["cargo" "--version"]
+  ["rustc" "--edition" "2021" "test.rs"]
+]
+depexts: [
+  ["cargo"] {os-distribution = "centos" & os-version >= "8"}
+  ["cargo"] {os-distribution = "ol" & os-version >= "8"}
+  ["cargo"] {os-distribution = "fedora"}
+  ["cargo"] {os-family = "suse"}
+  ["cargo"] {os-family = "debian"}
+  ["cargo"] {os-family = "ubuntu"}
+  ["cargo" "rustc"] {os-distribution = "nixos"}
+  ["cargo"] {os-family = "alpine"}
```

Comment:
```suggestion
  ["cargo"] {os-family = "alpine" & os-version > "3.15"}
```

Reply:
I am not sure about these. This package [compiles fine for Alpine 3.15](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/a5fc3b58a46d47eb8a9e75ad2ac3d03f0819e565/variant/distributions,alpine-3.15,conf-rust-2021.1). However, the change you're proposing means that no depext will be suggested for Alpine prior to 3.16. Is that what you want to achieve?

Reply:
If the package is available on 3.15 then my suggestion is wrong indeed.

And indeed, looking at the errors, it's a different one for 3.14 and 3.15:

> alpine-3.14
> 
>     [conf-rust-2021.1 (failed: edition 2021 is unstable and only available with -Z unstable-options.)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/a5fc3b58a46d47eb8a9e75ad2ac3d03f0819e565/variant/distributions,alpine-3.14,conf-rust-2021.1)
>     [tezos-rust-libs.1.2 (failed: edition 2021 is unstable and only available with -Z unstable-options.)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/a5fc3b58a46d47eb8a9e75ad2ac3d03f0819e565/variant/distributions,alpine-3.14,tezos-rust-libs.1.2)
> 
> alpine-3.15
> 
>     [tezos-rust-libs.1.2 (failed: package `corosensei v0.1.3` cannot be built because it requires rustc 1.59.0 or newer, while the currently active rustc version is 1.56.1)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/a5fc3b58a46d47eb8a9e75ad2ac3d03f0819e565/variant/distributions,alpine-3.15,tezos-rust-libs.1.2)

So it should be `>= "3.15"`

Same remark for `"ol"`.

Reply:
You might need to use an additional field in the `opam` file: `available:`. That additional field basically makes the package available/not for different platforms/constraints. That field should be used if e.g., the underlying system package cannot be installed (automatically by opam? at all?) on the specified platform.

I was trying to suggest fixes so that the package is simply not available on those platform that it is incompatible with. But I might have gotten wrong advice. The `available:` field is probably more important here.

I'm not very knowledgeable on `conf-`/system packages so we might want some additional info from maintainers.